src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -33,7 +31,7 @@ public struct ReadableBuffer : ISequence<ReadOnlyMemory<byte>>
         /// <summary>
         /// Determines if the <see cref="ReadableBuffer"/> is empty.
         /// </summary>
-        public bool IsEmpty => Length == 0;
+        public bool IsEmpty => _first.IsEmpty && Length == 0;
```

Comment:
I think we just need to prevent this scenario from happening or verify that it does not happen (you should never be left with an empty block at the beginning of a pipe)